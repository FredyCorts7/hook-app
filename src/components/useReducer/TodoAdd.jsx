import React from 'react';
import { useForm } from '../../hooks/useForm';

const TodoAdd = ({ handleAddTodo }) => {
  const [formValues, handleInputChange, reset] = useForm({
    title: '',
    description: '',
  });

  const handleSubmit = (e) => {
    e.preventDefault();

    const newTodo = {
      id: new Date().getTime(),
      title,
      description,
      done: false,
    };

    handleAddTodo(newTodo);
    reset();
  };

  const { title, description } = formValues;

  return (
    <form onSubmit={handleSubmit} className='container'>
      <h3 className='text-center'>Add ToDo</h3>

      <input
        type='text'
        name='title'
        placeholder='ex. learn ...'
        className='form-control'
        autoComplete='off'
        value={title}
        onChange={handleInputChange}
      />

      <input
        type='text'
        name='description'
        placeholder='ex. about that ...'
        className='form-control mt-2'
        autoComplete='off'
        value={description}
        onChange={handleInputChange}
      />

      <button type='submit' className='btn btn-outline-success btn-block mt-2'>
        Add ToDo
      </button>
    </form>
  );
};

export default TodoAdd;
