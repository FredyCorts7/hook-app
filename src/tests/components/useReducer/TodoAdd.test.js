import { shallow } from 'enzyme';
import React from 'react';
import TodoAdd from '../../../components/useReducer/TodoAdd';
import '@testing-library/jest-dom';

describe('Testing to <TodoAdd />', () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test('should show succesfully', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should not call handleAddTodo', () => {
    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test('should have been called handleAddTodo function', () => {
    const title = 'learn Nest.js';
    const description = 'Framework de Node.js';

    wrapper
      .find('input')
      .at(0)
      .simulate('change', {
        target: {
          value: title,
          name: 'title',
        },
      });

    wrapper
      .find('input')
      .at(1)
      .simulate('change', {
        target: {
          value: description,
          name: 'description',
        },
      });

    const formSubmit = wrapper.find('form').prop('onSubmit');

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    // expect(handleAddTodo).toBeCalledWith(expect.any(Object)); // {}
    expect(handleAddTodo).toBeCalledWith({
      id: expect.any(Number),
      title,
      description,
      done: false,
    });
    expect(wrapper.find('input').at(0).prop('value')).toBe('');
    expect(wrapper.find('input').at(1).prop('value')).toBe('');
  });
});
