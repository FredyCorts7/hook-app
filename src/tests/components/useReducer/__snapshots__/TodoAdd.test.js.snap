// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <TodoAdd /> should show succesfully 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <h3
    className="text-center"
  >
    Add ToDo
  </h3>
  <input
    autoComplete="off"
    className="form-control"
    name="title"
    onChange={[Function]}
    placeholder="ex. learn ..."
    type="text"
    value=""
  />
  <input
    autoComplete="off"
    className="form-control mt-2"
    name="description"
    onChange={[Function]}
    placeholder="ex. about that ..."
    type="text"
    value=""
  />
  <button
    className="btn btn-outline-success btn-block mt-2"
    type="submit"
  >
    Add ToDo
  </button>
</form>
`;
